lib/Doctrine/ORM/Query/Expr/Select.php

```diff
@@ -11,20 +11,17 @@
  */
 class Select extends Base
 {
-    /** @var string */
-    protected $preSeparator = '';
-
-    /** @var string */
-    protected $postSeparator = '';
+    protected string $preSeparator  = '';
+    protected string $postSeparator = '';
 
     /** @var string[] */
-    protected $allowedClasses = [Func::class];
+    protected array $allowedClasses = [Func::class];
```

Comment:
I don't know if that belongs into the scope (or if it fits here): `class-string[]` would be a nice addition.

Reply:
That's something that should be done on 2.14.x

Reply:
And IIRC, it's a rabbit hole if you want to specify something inside the `<>` of `class-string<>` :stuck_out_tongue:

Reply:
Let's revisit this in 2.14 again.